src/test/java/com/mentorship/vineservice/repository/VineRepositoryTest.java

```diff
@@ -0,0 +1,186 @@
+package com.mentorship.vineservice.repository;
+
+import static com.mentorship.vineservice.specification.VineSpecification.equalsColor;
+import static com.mentorship.vineservice.specification.VineSpecification.equalsGrape;
+import static com.mentorship.vineservice.specification.VineSpecification.equalsName;
+import static com.mentorship.vineservice.specification.VineSpecification.equalsSugar;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.mentorship.vineservice.domain.Vine;
+import com.mentorship.vineservice.dto.enums.VineColor;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@DataJpaTest
+public class VineRepositoryTest {
+
+    @Autowired
+    private VineRepository vineRepository;
+    private Vine merlotVine1;
+    private Vine zinfandelVine1;
+    private Vine cabernetVine2;
+    private Vine cabernetVine1;
+    private Vine merlotVine2;
+    private Vine merlotVine3;
+    private Map<Vine, Vine> vineMap;
+    private List<Vine> defaultListOfVines;
+
+
+    @BeforeEach
+    public void setUpData() {
+
+        merlotVine1 = createVineObject("MERLOT_VINE", VineColor.RED, "MERLOT", "SWEET");
+        zinfandelVine1 = createVineObject("ZINFANDEL_VINE", VineColor.ORANGE, "ZINFANDEL", "DRY");
+        cabernetVine2 = createVineObject("CABERNET_VINE", VineColor.WHITE, "CABERNET SAUVIGNON", "SWEET");
+        cabernetVine1 = createVineObject("CABERNET_VINE", VineColor.RED, "CABERNET SAUVIGNON", "SEMI-SWEET");
+        merlotVine2 = createVineObject("MERLOT_VINE", VineColor.WHITE, "MERLOT", "SWEET");
+        merlotVine3 = createVineObject("MERLOT_VINE", VineColor.ROSE, "MERLOT", "DRY");
+
+        defaultListOfVines = List.of(
+            merlotVine1,
+            zinfandelVine1,
+            cabernetVine2,
+            cabernetVine1,
+            merlotVine2,
+            merlotVine3
+        );
+
+        vineMap = new HashMap<>();
+
+        defaultListOfVines.forEach(vineObject -> {
+
+            Vine createdVine = vineRepository.save(vineObject);
+
+            vineMap.put(vineObject, createdVine);
+        });
+
+    }
+
+
+    @Test
+    public void shouldCreateVineAndReturnVineObject() {
+
+        Vine vineToSave = createVineObject("CHARDONNAY_VINE", VineColor.RED, "Chardonnay", "SWEET");
+
+        Vine createdVine = vineRepository.save(vineToSave);
+
+        Vine expectedVine = vineRepository.findById(createdVine.getId()).get();
```

Comment:
let's check here if vine exists by id and then compare it with object found by id